src/Native/FactorizationMachineNative/FactorizationMachineCore.cpp

```diff
@@ -0,0 +1,163 @@
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <pmmintrin.h>
+
+#define UNUSED(x) (void)(x)
+#define DEBUG_ONLY(x) (void)(x)
+
+#ifdef COMPILER_GCC
+
+#include "UnixSal.h"
+#define EXPORT_API(ret) extern "C" __attribute__((visibility("default"))) ret
+
+#else
+#include <intrin.h>
+#define EXPORT_API(ret) extern "C" __declspec(dllexport) ret __stdcall
+#endif
+
+EXPORT_API(void) CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, _In_ int * fieldIndices, _In_ int * featureIndices, _In_ float * featureValues,
```

Comment:
>Native [](start = 47, length = 6)

Native in the name of the function is kinda excessive

Reply:
Keeping it for readability in the other files, when they are invoked. There are non-native functions in the faff interface, that have the same signatures, and call into those. 

---
In reply to: [196861195](https://github.com/dotnet/machinelearning/pull/383#discussion_r196861195) [](ancestors = 196861195)